api/modules/budgeting.js

```diff
@@ -65,6 +89,18 @@ const budgeting = module.exports = (() => {
                 external_uuid_type: 'STRIPE'
             }
         })
+        const client = await clientManagement.findClientWithExternalId({
+            id: stripeInvoice.customer
+        })
+
+        try {
+            await checkExistingPaymentsWithExternalId({ id: stripeInvoice.id })
```

Comment:
Store the return value as a variable

```
const paymentsDoNotMatch = await checkForMismatchedClientPayments(...)
```

OR 

use a `throw` in the `checkForMismatchedClientPayments` function